homeassistant/helpers/event_decorators.py

```diff
@@ -0,0 +1,81 @@
+""" Event Decorators for custom components """
+
+import functools
+
+from homeassistant.helpers import event
+
+HASS = None
+
+
+def _callback(action, *args, **kwargs):
+    """ adds HASS to callback arguments """
+    action(HASS, *args, **kwargs)
+
+
+def track_state_change(entity_ids, from_state=None, to_state=None):
+    """ Decorator factory to track state changes for entity id """
+
+    def track_state_change_decorator(action):
+        """ Decorator to track state changes """
+        event.track_state_change(HASS, entity_ids,
+                                 functools.partial(_callback, action),
```

Comment:
Actually, thinking about this, wouldn't the following yield exactly the same result:
```python
functools.partial(action, HASS)
```